comm/src/main/scala/coop/rchain/comm/rp/HandleMessages.scala

```diff
@@ -75,16 +75,21 @@ object HandleMessages {
       _ <- ConnectionsCell[F].flatModify(_.addConnAndReport[F](peer))
     } yield handledWithoutMessage
 
-  def handleProtocolHandshake[F[_]: Monad: TransportLayer: Log: ErrorHandler: ConnectionsCell: RPConfAsk: Metrics](
+  def handleProtocolHandshake[F[_]: Monad: TransportLayer: Log: ConnectionsCell: RPConfAsk: Metrics](
       peer: PeerNode,
       protocolHandshake: ProtocolHandshake
   ): F[CommunicationResponse] =
     for {
       conf     <- RPConfAsk[F].ask
       response = ProtocolHelper.protocolHandshakeResponse(conf.local, conf.networkId)
-      _        <- TransportLayer[F].send(peer, response) >>= ErrorHandler[F].fromEither
-      _        <- Log[F].info(s"Responded to protocol handshake request from $peer")
-      _        <- ConnectionsCell[F].flatModify(_.addConnAndReport[F](peer))
+      resErr   <- TransportLayer[F].send(peer, response)
+      _ <- resErr.fold(
+            kp(().pure[F]),
+            kp(
+              Log[F].info(s"Responded to protocol handshake request from $peer") >>
+                ConnectionsCell[F].flatModify(_.addConnAndReport[F](peer))
+            )
```

Comment:
I'm sure we can have it simpler than `.fold(kp(().pure[F]), X)`. I think `.map(X)`?

Reply:
that gived you back an Either[, X] as a result  which is not what we want, right?

Reply:
right, that won't do. Feels awkwardly over-verbose, but having thought more of it, I don't see a way to make it simpler ;/